app/controllers/discourse_assign/assign_controller.rb

```diff
@@ -114,8 +115,75 @@ def assigned
       render json: { topics: serialize_data(topics, AssignedTopicSerializer) }
     end
 
+    def group_assignments
+      if SiteSetting.assign_enabled?
```

Comment:
Rather than wrapping this all in a huge `if` statement, you could do

```
raise Discourse::InvalidAccess.new if !SiteSetting.assign_enabled?
```
(Also, maybe we should use Discourse::NotFound, since 404 is a better error code for this situation)

Reply:
Changed as specified😀